addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/AccountHandler.java

```diff
@@ -390,27 +393,51 @@ private void refreshData() {
                 updateDeviceList();
                 updateFlashBriefingHandlers();
 
-                // update notification states
                 DeviceNotificationState[] deviceNotificationStates = null;
+                AscendingAlarmModel[] ascendingAlarmModels = null;
+                JsonBluetoothStates states = null;
+                List<JsonMusicProvider> musicProviders = null;
                 if (currentConnection.getIsLoggedIn()) {
+                    // update notification states
                     deviceNotificationStates = currentConnection.getDeviceNotificationStates();
-                }
 
-                // update ascending alarm
-                AscendingAlarmModel[] ascendingAlarmModels = null;
-                if (currentConnection.getIsLoggedIn()) {
+                    // update ascending alarm
                     ascendingAlarmModels = currentConnection.getAscendingAlarm();
-                }
 
-                // update bluetooth states
-                JsonBluetoothStates states = null;
-                if (currentConnection.getIsLoggedIn()) {
+                    // update bluetooth states
                     states = currentConnection.getBluetoothConnectionStates();
-                }
 
+                    // update music providers
+                    if (currentConnection.getIsLoggedIn()) {
+                        try {
+                            musicProviders = currentConnection.getMusicProviders();
+                        } catch (HttpException | JsonSyntaxException | ConnectionException e) {
+                            logger.debug("Update music provider failed {}", e);
+                        }
+                    }
+                }
                 // forward device information to echo handler
                 for (EchoHandler child : echoHandlers) {
                     Device device = findDeviceJson(child);
+
+                    @Nullable
```

Comment:
This `@Nullable` annotation is not needed.

Reply:
If I remove this, I got a validation error in for the usage in line 467.

Reply:
Okay, I see. Really confusing. Maybe @maggu2810 has an idea. If not we should leave it like it is.

Reply:
I assume you need to differ if the array can be null or if the array can contain nullable entries.

Reply:
Perhaps this could be useful for you about "nullness" and "array"

```Java
    public void fooReqReq(@NonNull String @NonNull [] foo) {
        foo[0] = null; // nullness violation
        foo = null; // nullness violation
    }

    public void fooReqOpt(@NonNull String @Nullable [] foo) {
        foo[0] = null; // nullness violation
        foo = null; // nullness handling okay
    }

    public void fooOptReq(@Nullable String @NonNull [] foo) {
        foo[0] = null; // nullness handling okay
        foo = null; // nullness violation
    }

    public void fooOptOpt(@Nullable String @Nullable [] foo) {
        foo[0] = null; // nullness handling okay
        foo = null; // nullness handling okay
    }
```

Reply:
So, I expect everything is correct. In my case, the array and the members of the array can be null. Local variables are default nullable, so there is no annotation needed for the array, but it is needed for the members.